### PR TITLE
MetadataTools.hasUuid returns false if entity extends MappedSuperclass

### PR DESCRIPTION
**Pull request by Gavrilov-Ivan**
See #1774

### CODE REVIEW

File: jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java [hunk 1]
```diff
@@ -869,12 +869,44 @@ public List<String> getSoftDeleteProperties(Class<?> clazz) {
...
+     * @return uuid property name or null if it doesn't exist
+     */
+    @Nullable
+    public String getUuidPropertyName(MetaClass metaClass) {
+        String uuidProperty = (String) metaClass.getAnnotations().get(UUID_KEY_ANN_NAME);
+        if(uuidProperty != null) {
+            return uuidProperty;
+        } else {
+            MetaClass ancestor = metaClass.getAncestor();
+            while (ancestor != null) {
```
Comment (gorbunkov) [inline jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java:901]: As I see there is a `io.jmix.core.metamodel.model.MetaClass#getAncestors` method that returns a full list of metaclass ancestors. No need to do a while loop here if you use this method.

File: jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java [hunk 2]
```diff
@@ -869,12 +869,44 @@ public List<String> getSoftDeleteProperties(Class<?> clazz) {
...
+    }
+
+    /**
+     * Gets name of uuid key property.
+     *
+     * @param metaClass entity MetaClass
+     * @return uuid property name or null if it doesn't exist
+     */
+    @Nullable
+    public String getUuidPropertyName(MetaClass metaClass) {
```
Comment (gorbunkov) [inline jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java:895]: Do we really need to introduce a new `getUuidPropertyName` method? Why not to implement the behavior in the existing `getUuidPropertyName(Class<?> clazz)`?

File: jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java [hunk 3]
```diff
@@ -869,12 +869,44 @@ public List<String> getSoftDeleteProperties(Class<?> clazz) {
...
-        return metaClass.getAnnotations().containsKey(UUID_KEY_ANN_NAME);
+        String uuidPropertyName = getUuidPropertyName(metaClass);
+        return uuidPropertyName != null;
     }
 
+    /**
+     * Gets name of uuid key property.
+     *
+     * @param clazz entity java class
+     * @return uuid property name or null if it doesn't exist
```
Comment (gorbunkov) [inline jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java:880]: After your changes the method will throw an exception if there is no metaClass for the given java class. It is worth mentioning in the javadoc.

**Review by gorbunkov — CHANGES_REQUESTED** [review]
It would be nice to have a test for the fixed issue

File: jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java (hunk 2 above)
Comment (Gavrilov-Ivan) [reply jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java:895]: To iliminate double conversion `MetaClass` -> `JavaClass` -> `MetaClass`:
`hasUuid` accepts `MetaClass` and uses result of `getUuidPropertyName` - original one accepts `Class<?>` and convert it to `MetaClass`. Not a critical point, we can keep this conversion and remove extra method.

File: jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java (hunk 3 above)
Comment (Gavrilov-Ivan) [reply jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java]: Done

File: jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java (hunk 1 above)
Comment (Gavrilov-Ivan) [reply jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java]: Originally done the same way as `getPrimaryKeyName`.
Rewritten with `getAncestors`.

**Comment by Gavrilov-Ivan** [comment]
Added test

**Review by gorbunkov — APPROVED** [review]

**Pull request merged**